learn/advanced/pagination.md

```diff
@@ -0,0 +1,117 @@
+```
+
+Once a query returns less than your configured `limit`, you have reached the last results page.
+
+It is often helpful to disable navigation buttons when the user cannot move to the "next" or "previous" page. The "previous" button should be disabled whenever your `offset` is 0.
+
+To disable the "next" button, we recommend setting your query's `limit` to the number of displayed results plus one. The extra result indicates that you have at least one item beyond what you can display in a single results page:
+
+```js
+const results = await index.search("x", { limit: 21, offset: 0 })
+
+if (offset === 0 ) {
+  // If offset equals 0, we're on the first results page
+  // and can disable the "previous" buttons
+  document.querySelector("#previous_button").disabled = true;
+} else {
+  document.querySelector("#previous_button").disabled = false;
+}
+
+if (results.hits.length < 21 ) {
+  // If Meilisearch returns 20 items or less, 
+  // we don't have any more results to 
+  // and can disable the "next"
+  document.querySelector("#next_button").disabled = true;
+} else {
+  // If Meilisearch returns exactly 21 results
+  // and our page can only show 20 items at a time,
+  // we have at least one more page with 1 result in it
+  document.querySelector("#next_button").disabled = false;
+}
+```
+
+## Not recommended: Page selection
+
+As its name states, this type of pagination is made of a numbered list of pages accompanied by next and previous buttons.
+
+![placeholder page selection UI](https://vuejsexamples.com/content/images/2018/11/vue-pagination.gif)
+
+This is is a common UI pattern that offers users a significant amount of precision when navigating results.
+
+### Implementation
+
+The general implementation of a page selection interface is similar to our [recommended solution](#recommended-previous-and-next-buttons), but you must also display a page list.
+
+To create the page list, however, you must know the exact number of total results. For example, if you have 100 results and your search result pages contain 10 results each, your selector must show a list of numbers going from 1 to 10.
+
+Since Meilisearch can only give you an estimate of total search results, it can be difficult to implement page selectors when using Meilisearch.
+
+We recommend two different workarounds to create this kind of pagination interface with Meilisearch.
+
+#### Set a hard limit of results during your search
+
+By default, a search request returns 20 search results. You can change this value to a much higher number and treat it as the effective maximum of search results you will show a user. Doing so means the size of your `hits` array is the exact number of search results you have to paginate.
+
+For example, if you set `limit` to 300, every search request made to Meilisearch returns at most 300 documents. If a query returns a `hits` array with 200 items and you want each page to display 20 results, you can create a page selector with 10 pages.
```

Comment:
This seems like a very costly method since you are retrieving the full number of search results regardless of whether the user proceeds to the next page. Wouldn't it be better to use the new instance setting `maxTotalHits`, and keep using `limit` for the number of results per page?

Reply:
I'm not sure I understand your question @dichotommy.

`maxTotalHits` does not affect the calculation at the time of the search. It just prevents going further than this `maxTotalHits` value when browsing the search results.

Using `limit` at `200` for example forces the engine to be "exhaustive" for the calculation of the 200 results and therefore allows to generate the page selector for these 200 results. Each time you will switch a page, it does not emit a new request because it has been cached on the browser side. Pros. It's fast for any other page than the first one. Cons. You have to choose a high `limit`, it's counter-intuitive and not clean, and slow to render the search results for the first call. 

It's clear that it's expensive and bad for the performance. Hence the fact that this method by page selection is also not recommended.

Reply:
Thanks for the explanation, @gmourier 👍 

I agree with you: this section should stay. I might change the text slightly so it's clear that `limit` effectively forces an "exhaustive" search (within the configured value, ofc).